.gitlab/package_build/linux.yml

```diff
@@ -80,6 +80,14 @@
   before_script:
     - export RELEASE_VERSION=$RELEASE_VERSION_7
 
+.agent_fips_build:
+  variables:
+    AGENT_MAJOR_VERSION: 7
+    FLAVOR: fips
+  before_script:
+    - export RELEASE_VERSION=$RELEASE_VERSION_7
+    - DD_GO_TOOLCHAIN="msgo" source /root/.bashrc
```

Comment:
can the same results be achieved by having `DD_GO_TOOLCHAIN` as a regular gitlab variable, and letting `source /root/.bashrc` be sourced by the runner?

What I'm getting at is that I fear we have too many `before_script` to easily reason about them. (that being said we probably could get rid of most since they are mostly about dispatching the agent version between 6 and 7)

Reply:
> letting `source /root/.bashrc` be sourced by the runner?

do you know at what point /root/.bashrc is sourced by the runner? 

this envvar modifies [the PATH](https://github.com/DataDog/datadog-agent-buildimages/blob/main/setup_go.sh#L37-L43) so if it is exported in the runner at initialization I think that will work but I'm not the most familiar with the timing of this, will try it 👍

Reply:
https://github.com/DataDog/datadog-agent/pull/31004#discussion_r1850065801

alex suggests we do not source .bashrc by default in runners so I believe this is still necessary

Reply:
fixed in https://github.com/DataDog/datadog-agent/pull/31004/commits/dada6cbb431637c499d1407f1c43bb5a7b70baaf